app.rb

```diff
@@ -0,0 +1,146 @@
+require_relative './student'
+require_relative './teacher'
+require_relative './book'
+require_relative './rental'
+
+module App
+  module PeopleListing
+    def list_all_peoples(peoples)
+      if peoples.empty?
+        puts "\n NO PERSON REGISTERED!"
+      else
+        puts "\n"
+        peoples.each do |people|
+          puts "ID: #{people.id}"
+          puts "NAME:  #{people.name}"
+          puts "AGE: #{people.age}"
+          puts '========================'
+          puts
+        end
+      end
+    end
+
+    def create_person(person)
+      print "\n Do you want to create a  Student (1) or Teacher (2)? [Input the number]: "
+      option = gets.chomp
+      case option
+      when '1'
+        create_student(person)
+      when '2'
+        create_teacher(person)
+      end
+    end
+
+    def create_student(person)
+      puts "\n"
+      print 'Age: '
+      age = gets.chomp
+      print 'Name: '
+      name = gets.chomp
+      print 'Has Parent Permission? [Y/N]: '
+      permission = gets.chomp.upcase
+      print 'Classroom: '
+      classroom = gets.chomp
+
+      parent_permission = permission == 'Y'
+
+      student = Student.new(classroom, age, name, parent_permission: parent_permission)
+      person.push(student)
+      puts "\n Student created successfully!!"
+    end
+
+    def create_teacher(person)
+      print 'Age: '
+      age = gets.chomp
+      print 'Name: '
+      name = gets.chomp
+      print 'Specialization: '
+      specialization = gets.chomp
+
+      teacher = Teacher.new(age, specialization, name)
+
+      person.push(teacher)
+
+      puts 'Teacher created successfully!!'
+    end
+  end
+
+  module BookLisitng
+    def list_all_books(books)
+      if books.empty?
+        puts "\n NO BOOKS REGISTERED!"
+      else
+        puts "\n"
+        books.each do |book|
+          puts "Title: #{book.title}"
+          puts "Author #{book.author}"
+          puts '========================'
+          puts
```

Comment:
- [ ] After creating a book, and trying to list all books, the app crashes with errors shown in the screenshot below in displaying the book's author. Kindly fix that 👍🏽

<img width="1304" alt="Screenshot 2023-04-20 at 10 37 20 PM" src="https://user-images.githubusercontent.com/50941074/233502022-79362aef-ff8e-4694-bf4c-6ba0ee18a069.png">

Reply:
Ok, thanks, I will fix